bundles/org.openhab.binding.enigma2/src/main/java/org/openhab/binding/enigma2/internal/Enigma2Client.java

```diff
@@ -0,0 +1,336 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.enigma2.internal;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.util.UrlEncoded;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.IOException;
+import java.io.StringReader;
+import java.time.LocalDateTime;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * The {@link Enigma2Client} class is responsible for communicating with the Enigma2 device.
+ * @see <a href="https://github.com/E2OpenPlugins/e2openplugin-OpenWebif/wiki/OpenWebif-API-documentation">OpenWebif-API-documentation</a>
+ *
+ * @author Guido Dolfen - Initial contribution
+ */
+@NonNullByDefault
+public class Enigma2Client {
+    private final Logger logger = LoggerFactory.getLogger(Enigma2Client.class);
+
+    static final String PATH_REMOTE_CONTROL = "/web/remotecontrol?command=";
+    static final String PATH_POWER = "/web/powerstate";
+    static final String PATH_VOLUME = "/web/vol";
+    static final String PATH_SET_VOLUME = "/web/vol?set=set";
+    static final String PATH_TOGGLE_MUTE = "/web/vol?set=mute";
+    static final String PATH_TOGGLE_POWER = "/web/powerstate?newstate=0";
+    static final String PATH_MESSAGE = "/web/message?type=";
+    static final String PATH_ALL_SERVICES = "/web/getallservices";
+    static final String PATH_ZAP = "/web/zap?sRef=";
+    static final String PATH_CHANNEL = "/web/subservices";
+    static final String PATH_EPG = "/web/epgservicenow?sRef=";
+    static final String PATH_ANSWER = "/web/messageanswer?getanswer=now";
+    static final int TYPE_QUESTION = 0;
+    static final int TYPE_INFO = 1;
+    static final int TYPE_WARNING  = 2;
+    static final int TYPE_ERROR = 3;
+    private Map<String,@Nullable String> channels = new LinkedHashMap<>();
+    private String host;
+    private boolean power;
+    private String channel = "";
+    private String title = "";
+    private String description = "";
+    private String answer = "";
+    private int volume = 0;
+    private int timeout;
```

Comment:
can this be made final?
Also this field should be renamed to reduce confusion between it and `timeout` parameters in other methods.

Reply:
Fixed